### ISSUE TITLE
run.pl 1/32 Failed

### ISSUE THREAD

**Issue by Hitori-de** [espnet/espnet#6049]
I was running the asr2 librispeech recipe for espnet2 and I ran the run.sh command then after a while it stopped and said that run.pl 1/32 failed. below is the complete output.

`(espnet) lwobole@lwobole:~/Thesis/espnet/egs2/librispeech/asr2$ ./run.sh --stage 4 --stop_stage 15
2025-02-24T15:38:32 (asr2.sh:288:main) ./asr2.sh --kmeans_opts --batch_bins 1200000 --kmeans_feature wavlm_large/21 --nclusters 2000 --ngpu 2 --src_lang wavlm_large_21_km2000 --tgt_lang en --src_token_type bpe --src_nbpe 6000 --tgt_token_type bpe --tgt_nbpe 5000 --src_case rm --tgt_case ts --speed_perturb_factors 0.9 1.0 1.1 --asr_config conf/tuning/train_discrete_asr_e_branchformer1.yaml --inference_config conf/decode_ctc0.3.yaml --train_set train_960 --valid_set dev --test_sets test_clean test_other dev_clean dev_other --src_bpe_train_text data/train_960/text.rm.wavlm_large_21_km2000 --tgt_bpe_train_text data/train_960/text.ts.en --lm_train_text data/train_960/text.ts.en data/local/other_text/text --stage 4 --stop_stage 15
2025-02-24T15:38:32 (asr2.sh:625:main) Skipped stages:  11 16 17 
2025-02-24T15:38:32 (asr2.sh:729:main) Stage 4: Remove long/short data: dump/audio_raw/org -> dump/audio_raw
utils/copy_data_dir.sh: copied data from dump/audio_raw/org/train_960 to dump/audio_raw/train_960
utils/validate_data_dir.sh: Successfully validated data-directory dump/audio_raw/train_960
fix_data_dir.sh: kept all 281241 utterances.
fix_data_dir.sh: old files are kept in dump/audio_raw/train_960/.backup
filter samples: removed 0 samples with empty text
utils/copy_data_dir.sh: copied data from dump/audio_raw/org/train_960_sp0.9 to dump/audio_raw/train_960_sp0.9
utils/validate_data_dir.sh: Successfully validated data-directory dump/audio_raw/train_960_sp0.9
fix_data_dir.sh: kept 281238 utterances out of 281241
fix_data_dir.sh: old files are kept in dump/audio_raw/train_960_sp0.9/.backup
filter samples: removed 3 samples with empty text
utils/copy_data_dir.sh: copied data from dump/audio_raw/org/train_960_sp1.1 to dump/audio_raw/train_960_sp1.1
utils/validate_data_dir.sh: Successfully validated data-directory dump/audio_raw/train_960_sp1.1
fix_data_dir.sh: kept all 281241 utterances.
fix_data_dir.sh: old files are kept in dump/audio_raw/train_960_sp1.1/.backup
filter samples: removed 0 samples with empty text
utils/copy_data_dir.sh: copied data from dump/audio_raw/org/dev to dump/audio_raw/dev
utils/validate_data_dir.sh: Successfully validated data-directory dump/audio_raw/dev
fix_data_dir.sh: kept 5551 utterances out of 5567
fix_data_dir.sh: old files are kept in dump/audio_raw/dev/.backup
filter samples: removed 16 samples with empty text
2025-02-24T15:39:01 (asr2.sh:779:main) Stage 5a: Perform Kmeans using wavlm_large features
2025-02-24T15:39:01 (perform_kmeans.sh:55:main) scripts/feats/perform_kmeans.sh --stage 1 --stop-stage 4 --train_set train_960 --dev_set dev --other_sets test_clean test_other dev_clean dev_other  train_960_sp0.9 train_960_sp1.1  --datadir dump/audio_raw --featdir dump/extracted --audio_format flac --feature_type wavlm_large --layer 21 --feature_conf {type=s3prl,conf={s3prl_conf={upstream=wavlm_large},download_dir=ckpt,multilayer_feature=False,layer=21}} --km_dir exp/kmeans/wavlm_large_21_2000clusters --portion 0.1 --nclusters 2000 --storage_save_mode true --use_gpu true --nj 32 --cpu_cmd run.pl --cuda_cmd run.pl --batch_bins 1200000
2025-02-24T15:39:01 (perform_kmeans.sh:93:main) stage 1: Dump wavlm_large feature
utils/subset_data_dir.sh: reducing #utt from 281241 to 28124
2025-02-24T15:39:02 (perform_kmeans.sh:120:main) Subsampling 28124 utterances for feature dumping.
Dump SSL train_960_subset0.1 features to dump/extracted/wavlm_large/layer21/train_960_subset0.1
utils/copy_data_dir.sh: copied data from dump/audio_raw/train_960_subset0.1 to dump/extracted/wavlm_large/layer21/train_960_subset0.1
utils/validate_data_dir.sh: Successfully validated data-directory dump/extracted/wavlm_large/layer21/train_960_subset0.1
run.pl: 1 / 32 failed, log is in dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/dump_features.*.log
`

I checked the log files and this is what it said.

`# python3 pyscripts/feats/dump_ssl_feature.py --feature_conf '{type=s3prl,conf={s3prl_conf={upstream=wavlm_large},download_dir=ckpt,multilayer_feature=False,layer=21}}' --audio_sample_rate 16000 --use_gpu true --in_filetype sound --out_filetype mat --write_num_frames ark,t:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/utt2num_frames.32 --utt2num_samples dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/utt2num_samples.32 --batch_bins 1200000 scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/wav.32.scp ark,scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.ark,dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.scp 
 Started at Mon Feb 24 17:40:40 PST 2025
#
2025-02-24 17:40:42 | INFO | root | Namespace(audio_sample_rate=16000, batch_bins=1200000, feature_conf='{type=s3prl,conf={s3prl_conf={upstream=wavlm_large},download_dir=ckpt,multilayer_feature=False,layer=21}}', in_filetype='sound', out_filetype='mat', rspecifier='scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/wav.32.scp', use_gpu=True, utt2num_samples='dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/utt2num_samples.32', write_num_frames='ark,t:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/utt2num_frames.32', wspecifier='ark,scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.ark,dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.scp')
2025-02-24 17:40:42 | INFO | root | Loading Features
2025-02-24 17:40:42 | INFO | root | Feature configuration: {'type': 's3prl', 'conf': {'s3prl_conf': {'upstream': 'wavlm_large'}, 'download_dir': 'ckpt', 'multilayer_feature': 'False', 'layer': '21'}}
2025-02-24 17:40:45 | INFO | s3prl.util.download | Requesting URL: https://huggingface.co/s3prl/converted_ckpts/resolve/main/wavlm_large.pt
2025-02-24 17:40:45 | INFO | s3prl.util.download | Using URL's local file: ckpt/f2d5200177fd6a33b278b7b76b454f25cd8ee866d55c122e69fccf6c7467d37d.wavlm_large.pt
2025-02-24 17:40:45 | INFO | s3prl.upstream.wavlm.WavLM | WavLM Config: {'extractor_mode': 'layer_norm', 'encoder_layers': 24, 'encoder_embed_dim': 1024, 'encoder_ffn_embed_dim': 4096, 'encoder_attention_heads': 16, 'activation_fn': 'gelu', 'layer_norm_first': True, 'conv_feature_layers': '[(512,10,5)] + [(512,3,2)] * 4 + [(512,2,2)] * 2', 'conv_bias': False, 'feature_grad_mult': 1.0, 'normalize': True, 'dropout': 0.0, 'attention_dropout': 0.0, 'activation_dropout': 0.0, 'encoder_layerdrop': 0.0, 'dropout_input': 0.0, 'dropout_features': 0.0, 'mask_length': 10, 'mask_prob': 0.8, 'mask_selection': 'static', 'mask_other': 0.0, 'no_mask_overlap': False, 'mask_min_space': 1, 'mask_channel_length': 10, 'mask_channel_prob': 0.0, 'mask_channel_selection': 'static', 'mask_channel_other': 0.0, 'no_mask_channel_overlap': False, 'mask_channel_min_space': 1, 'conv_pos': 128, 'conv_pos_groups': 16, 'relative_position_embedding': True, 'num_buckets': 320, 'max_distance': 800, 'gru_rel_pos': True}
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/s3prl/upstream/byol_s/byol_a/common.py:20: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend("sox_io")
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/torch/nn/utils/weight_norm.py:28: UserWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.
  warnings.warn("torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.")
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/torch/nn/functional.py:5137: UserWarning: Support for mismatched key_padding_mask and attn_mask is deprecated. Use same type for both instead.
  warnings.warn(
**2025-02-24 17:44:31 | INFO | s3prl_feature_loader | finished successfully**
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/torch/nn/modules/conv.py:306: **UserWarning:** Plan failed with a cudnnException: CUDNN_BACKEND_EXECUTION_PLAN_DESCRIPTOR: cudnnFinalize Descriptor Failed cudnn_status: CUDNN_STATUS_NOT_SUPPORTED (Triggered internally at /opt/conda/conda-bld/pytorch_1712608851799/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:919.)
  return F.conv1d(input, weight, bias, self.stride,
Accounting: time=232 threads=1
Ended (code 0) at Mon Feb 24 17:44:32 PST 2025, elapsed time 232 seconds`

It said that it finished successfully then it shows a UserWarning.
Is this okay? or something is defintely wrong that I have to fix and execute the command and whole process again?

**Comment by sw005320** [comment]
It is an error, and it's better to identify it.
It runs 32 cpu jobs at the same time, and some system errors happen.
So, I recommend you reduce the number of jobs and rerun it.
If you still encounter the issue, there is a data issue.

**Comment by Hitori-de** [comment]
> It is an error, and it's better to identify it. It runs 32 cpu jobs at the same time, and some system errors happen. So, I recommend you reduce the number of jobs and rerun it. If you still encounter the issue, there is a data issue.

Oh, I see; okay, I will try to check on that. Thank you!

**Comment by Hitori-de** [comment]
> It is an error, and it's better to identify it. It runs 32 cpu jobs at the same time, and some system errors happen. So, I recommend you reduce the number of jobs and rerun it. If you still encounter the issue, there is a data issue.

I tried reducing the CPU jobs to 16 and ran the training again. However, it still shows that it failed.
It shows a cudnn failed error with cudnn exception and CUDNN status not supported.

below is the log file:

`# python3 pyscripts/feats/dump_ssl_feature.py --feature_conf '{type=s3prl,conf={s3prl_conf={upstream=wavlm_large},download_dir=ckpt,multilayer_feature=False,layer=21}}' --audio_sample_rate 16000 --use_gpu true --in_filetype sound --out_filetype mat --write_num_frames ark,t:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/utt2num_frames.32 --utt2num_samples dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/utt2num_samples.32 --batch_bins 1200000 scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/wav.32.scp ark,scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.ark,dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.scp 
# Started at Mon Feb 24 21:08:32 PST 2025
#
2025-02-24 21:08:33 | INFO | root | Namespace(audio_sample_rate=16000, batch_bins=1200000, feature_conf='{type=s3prl,conf={s3prl_conf={upstream=wavlm_large},download_dir=ckpt,multilayer_feature=False,layer=21}}', in_filetype='sound', out_filetype='mat', rspecifier='scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/wav.32.scp', use_gpu=True, utt2num_samples='dump/extracted/wavlm_large/layer21/train_960_subset0.1/logdir/utt2num_samples.32', write_num_frames='ark,t:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/utt2num_frames.32', wspecifier='ark,scp:dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.ark,dump/extracted/wavlm_large/layer21/train_960_subset0.1/data/feats.32.scp')
2025-02-24 21:08:33 | INFO | root | Loading Features
2025-02-24 21:08:33 | INFO | root | Feature configuration: {'type': 's3prl', 'conf': {'s3prl_conf': {'upstream': 'wavlm_large'}, 'download_dir': 'ckpt', 'multilayer_feature': 'False', 'layer': '21'}}
2025-02-24 21:08:38 | INFO | s3prl.util.download | Requesting URL: https://huggingface.co/s3prl/converted_ckpts/resolve/main/wavlm_large.pt
2025-02-24 21:08:38 | INFO | s3prl.util.download | Using URL's local file: ckpt/f2d5200177fd6a33b278b7b76b454f25cd8ee866d55c122e69fccf6c7467d37d.wavlm_large.pt
2025-02-24 21:08:39 | INFO | s3prl.upstream.wavlm.WavLM | WavLM Config: {'extractor_mode': 'layer_norm', 'encoder_layers': 24, 'encoder_embed_dim': 1024, 'encoder_ffn_embed_dim': 4096, 'encoder_attention_heads': 16, 'activation_fn': 'gelu', 'layer_norm_first': True, 'conv_feature_layers': '[(512,10,5)] + [(512,3,2)] * 4 + [(512,2,2)] * 2', 'conv_bias': False, 'feature_grad_mult': 1.0, 'normalize': True, 'dropout': 0.0, 'attention_dropout': 0.0, 'activation_dropout': 0.0, 'encoder_layerdrop': 0.0, 'dropout_input': 0.0, 'dropout_features': 0.0, 'mask_length': 10, 'mask_prob': 0.8, 'mask_selection': 'static', 'mask_other': 0.0, 'no_mask_overlap': False, 'mask_min_space': 1, 'mask_channel_length': 10, 'mask_channel_prob': 0.0, 'mask_channel_selection': 'static', 'mask_channel_other': 0.0, 'no_mask_channel_overlap': False, 'mask_channel_min_space': 1, 'conv_pos': 128, 'conv_pos_groups': 16, 'relative_position_embedding': True, 'num_buckets': 320, 'max_distance': 800, 'gru_rel_pos': True}
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/s3prl/upstream/byol_s/byol_a/common.py:20: UserWarning: torchaudio._backend.set_audio_backend has been deprecated. With dispatcher enabled, this function is no-op. You can remove the function call.
  torchaudio.set_audio_backend("sox_io")
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/torch/nn/utils/weight_norm.py:28: UserWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.
  warnings.warn("torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.")
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/torch/nn/functional.py:5137: UserWarning: Support for mismatched key_padding_mask and attn_mask is deprecated. Use same type for both instead.
  warnings.warn(
2025-02-24 21:12:28 | INFO | s3prl_feature_loader | finished successfully
/home/lwobole/Thesis/espnet/tools/miniconda/envs/espnet/lib/python3.8/site-packages/torch/nn/modules/conv.py:306: UserWarning: Plan failed with a cudnnException: CUDNN_BACKEND_EXECUTION_PLAN_DESCRIPTOR: cudnnFinalize Descriptor Failed cudnn_status: CUDNN_STATUS_NOT_SUPPORTED (Triggered internally at /opt/conda/conda-bld/pytorch_1712608851799/work/aten/src/ATen/native/cudnn/Conv_v8.cpp:919.)
  return F.conv1d(input, weight, bias, self.stride,
# Accounting: time=237 threads=1
# Ended (code 0) at Mon Feb 24 21:12:29 PST 2025, elapsed time 237 seconds`

**Comment by sw005320** [comment]
I think some audio files are broken or too short/long.
It is better to identify them and remove them.

**Comment by Hitori-de** [comment]
> I think some audio files are broken or too short/long. It is better to identify them and remove them.

I encountered a new run.pl error during training and this was the log file.

`# python3 pyscripts/utils/learn_kmeans.py --km_path exp/kmeans/wavlm_large_21_2000clusters/km_2000.mdl --n_clusters 2000 --percent -1 --in_filetype mat scp:exp/kmeans/wavlm_large_21_2000clusters/train.scp 
#
2025-02-27 00:35:55 | INFO | root | Namespace(batch_size=10000, in_filetype='mat', init='k-means++', km_path='exp/kmeans/wavlm_large_21_2000clusters/km_2000.mdl', max_iter=100, max_no_improvement=100, n_clusters=2000, n_init=20, percent=-1.0, reassignment_ratio=0.0, rspecifier=['scp:exp/kmeans/wavlm_large_21_2000clusters/train.scp'], seed=0, tol=0.0)
`
